src/loop_libraries/library_looptools.cpp

```diff
@@ -117,46 +115,30 @@ void Looptools::B(std::array<std::complex<double>, 2>& b, B_ARGS) noexcept
 
 void Looptools::C(std::array<std::complex<double>, 7>& c, C_ARGS) noexcept
 {
-   double p10 = p10_in.real();
-   double p21 = p21_in.real();
-   double p20 = p20_in.real();
-   double m02 = m02_in.real();
-   double m12 = m12_in.real();
-   double m22 = m22_in.real();
+   const int coeffs[] = {cc0, cc1, cc2, cc00, cc11, cc12, cc22};
+   ComplexType res[Ncc];
    set_mu2_uv(scl2_in);
-   c.at(0) = C0i(cc0, p10, p21, p20, m02, m12, m22);
-   c.at(1) = C0i(cc1, p10, p21, p20, m02, m12, m22);
-   c.at(2) = C0i(cc2, p10, p21, p20, m02, m12, m22);
-   c.at(3) = C0i(cc00, p10, p21, p20, m02, m12, m22);
-   c.at(4) = C0i(cc11, p10, p21, p20, m02, m12, m22);
-   c.at(5) = C0i(cc12, p10, p21, p20, m02, m12, m22);
-   c.at(6) = C0i(cc22, p10, p21, p20, m02, m12, m22);
+
+   Cput(res, p10_in.real(), p21_in.real(), p20_in.real(), m02_in.real(),
+        m12_in.real(), m22_in.real());
+   for (int i = 0; i < 7; ++i) {
```

Comment:
Would you mind replacing `7` by `sizeof(coeffs)/sizeof(coeffs[0])`? (Also below in the D function routine.) This would be a bit more secure/readable, since one connects this magic number to the array size.